### PR TITLE
refactor(jss-manager): refactor react JSS Manager to treat all styles as dynamic

### LINKED ISSUE #193: Increase code coverage for global branch to 80
"coverageThreshold": {
      "global": {
        **"branches": 69,**
        "functions": 80,
        "lines": 80,
        "statements": -10
        }
    },

### LINKED ISSUE #197: Fix "TODO" issue in packages/fast-jss-manager-react/src/utilities/get-static-styles.ts
TODO found

https://codeclimate.com/github/Microsoft/fast-dna/packages/fast-jss-manager-react/src/utilities/get-static-styles.ts#issue_5abfc3249ab05300010001da

### LINKED ISSUE #331: Update React JSS Manager to remove static stylesheet behavior and handle all styles as dynamic
### Specifications (hardware, software, configurations)
* Include device make, model, OS version: 
* Include browser and version: 
* Include screen orientation: 
* Include screen size: 
* Include page URL: 
* Include localization: 

*Please attach screenshots or video of the problem for quickest resolution*

### PR DESCRIPTION
**Pull request by chrisdholt**
closes #331, closes #197, closes #193
<body>

<footer>
 

For [details on formatting](https://github.com/Microsoft/fast-dna/wiki/pull-request-guidance) pull requests.

### CODE REVIEW

File: packages/fast-jss-manager-react/src/manage-jss.spec.tsx [hunk 1]
```diff
@@ -78,13 +74,13 @@ describe("The higher-order component", (): void => {
         expect(Component["stylesheetManager"].sheets.length).toBe(expected);
     });
     // tslint:disable-next-line
-    test("should share the static portion of a stylesheet and not share the dynamic portion of a stylesheets across component instances", (): void => {
+    test("should note share the static portion or the dynamic portion of a stylesheets across component instances", (): void => {
```
Comment (nicholasrice) [inline packages/fast-jss-manager-react/src/manage-jss.spec.tsx]: not

File: packages/fast-jss-manager-react/src/manage-jss.tsx [hunk 2]
```diff
@@ -95,9 +67,9 @@ function manageJss<S, C>(styles?: ComponentStyles<S, C>): <T>(Component: React.C
             private static styleMap: WeakMap<ComponentStyles<S, C>, string> = new WeakMap();
 
             /**
-             * Store references to all separated stylesheets
+             * All static and dynamic styles associated with the component
```
Comment (nicholasrice) [inline packages/fast-jss-manager-react/src/manage-jss.tsx]: They are actually the styles of *all* component instances - I'm not sure you'll actually need this anymore though

File: packages/fast-jss-manager-react/src/manage-jss.tsx [hunk 3]
```diff
@@ -109,44 +81,11 @@ function manageJss<S, C>(styles?: ComponentStyles<S, C>): <T>(Component: React.C
...
-                // Store the instance id as a combination of the generated IDs
-                this.instanceId = `${componentId}${styleId}`;
-                let separatedStylesInstance: ISeparatedStylesheet<S, C> = JSSManager.separatedStyles[this.instanceId];
-
-                // Check if we have a separatedStyles object stored at the instanceId key.
-                // If we don"t, we need to create that object
-                if (!Boolean(separatedStylesInstance)) {
-                    separatedStylesInstance = JSSManager.separatedStyles[this.instanceId] = this.separateStyles(styles);
-                }
+                JSSManager.styles = styles;
```
Comment (nicholasrice) [inline packages/fast-jss-manager-react/src/manage-jss.tsx]: Yea I don't think you need to store this anymore - you an just reference `styles` throughout.

With this implementation, you're actually clobbering a static value for every instance of a component

File: packages/fast-jss-manager-react/src/manage-jss.tsx [hunk 4]
```diff
@@ -109,44 +81,11 @@ function manageJss<S, C>(styles?: ComponentStyles<S, C>): <T>(Component: React.C
...
-                // object if they exist
-                const staticStyleSheet: StyleSheet = separatedStylesInstance.staticStyleSheet;
-                const dynamicStyles: StyleSheet = separatedStylesInstance.dynamicStyles;
-
-                if (Boolean(staticStyleSheet)) {
-                    state.staticStyleSheet = separatedStylesInstance.staticStyleSheet;
-                }
-
-                if (Boolean(dynamicStyles)) {
+                if (Boolean(JSSManager.styles)) {
```
Comment (nicholasrice) [inline packages/fast-jss-manager-react/src/manage-jss.tsx]: we should probably just rename to `styleSheet` - no reason to differentiate these as "dynamic" anymore

File: packages/fast-jss-manager-react/src/manage-jss.tsx [hunk 5]
```diff
@@ -232,56 +162,23 @@ function manageJss<S, C>(styles?: ComponentStyles<S, C>): <T>(Component: React.C
              */
             private createDynamicStyleSheet(): any {
                 return jss.createStyleSheet(
-                    merge({}, JSSManager.separatedStyles[this.instanceId].dynamicStyles, this.props.jssStyleSheet),
+                    merge({}, JSSManager.styles, this.props.jssStyleSheet),
```
Comment (nicholasrice) [inline packages/fast-jss-manager-react/src/manage-jss.tsx]: this just becomes `styles`

File: packages/fast-jss-manager-react/src/manage-jss.spec.tsx (hunk 1 above)
Comment (chrisdholt) [reply packages/fast-jss-manager-react/src/manage-jss.spec.tsx]: DANGIT - yep.

**Pull request merged**